### DISCUSSION TITLE
How can I create routes by roles with next-auth/authjs nextjs/13/14?

### DISCUSSION THREAD

**Discussion by metacop** [vercel/next.js#59884] [Help] [answered]
### Summary

how can I protect private routes by roles. That is, if I am a user type "user" I can have access to the /normal route and if I am admin I can have access to the /admin route. That is, multiple private routes with specific roles, the official documentation only shows an example of a single role for a single route. but what I want is to make it multiple.

```
//middleware.ts

import { withAuth } from "next-auth/middleware"

export default withAuth(
  // `withAuth` augments your `Request` with the user's token.
  function middleware(req) {
    console.log(req.nextauth.token)
  },
  {
    callbacks: {
      authorized: ({ token }) => token?.role === "admin",
    },
  }
)

export const config = { matcher: ["/admin"] }
```

Here is the code extracted from the official next-auth documentation https://next-auth.js.org/configuration/nextjs

what I want is that "export const config = { matcher: ['/dashboard'] }; " applies to that callback type "user" but I want to make more protected routes depending on the user's role

I am using nextjs 14

### Additional information

```js
nextjs next-auth authjs private route next14
```

**Comment by ananuness** [comment, marked as answer]
Considering that you are already making the role information available in the token, a possible solution is:

```js
import { withAuth } from "next-auth/middleware";
import { NextRequest, NextResponse } from "next/server";

export default withAuth(
  function middleware(req) {
    if (req.nextUrl.pathname.startsWith("/admin") && req.nextauth.token?.role !== "admin") {
      return NextResponse.rewrite(
        new URL("/auth/login?message=Not Authorized", req.url)
      );
    }

    if (req.nextUrl.pathname.startsWith("/user") && req.nextauth.token?.role !== "user") {
      return NextResponse.rewrite(
        new URL("/auth/login?message=Not Authorized", req.url)
      );
    }
  },
  {
    callbacks: {
      authorized: ({ token }) => !!token,
    },
  }
);

export const config = {
  matcher: ["/admin/:path*", "/user/:path*"],
};
```
So, considering this logic, it would be nice if your routes were well organized to make them easier to validate here.

**Reply by metacop** [reply]
thanks!
